### PR TITLE
Adding Sample Correlation Analysis

### PR DESCRIPTION
**Pull request by gwaygenomics**
Many files are added here (mostly figures). The main addition is the analysis in f2018d7 and updated functions in 23ec397. There is also a minor update to the README in 244a30b and environment in 4b84b16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ To reproduce the results of the analysis, the modules should be run in order.
 | 1.initial-z-sweep | Determine a set of optimal hyperparameters for Tybalt and ADAGE models across a representative range of z dimensionality |
 | 2.ensemble-z-analysis | Train various algorithms to compress gene expression data across a large range of z dimensions |
 | 3.build-hetnets | Download, process, and integrate various curated gene sets into a single heterogeneous network |
+| 4.analyze-components | Visualize the reconstruction and sample correlation results of the ensemble z analysis |
```
Comment (ypar) [inline README.md]: this README.md could benefit from some more detailed index/list of some subdirectory notebooks/scripts. for example: 

| 0.expression-download | Download and process gene expression data to run through pipeline |
| [0.1.A.download-pancanatlas-data.ipynb](hyperlink)|some description|
| [0.1.B...](some other hyperlink)|some desc|
| 1.initial-z-sweep | Determine a set of optimal hyperparameters for Tybalt and ADAGE models across a representative range of z dimensionality |
| [1.A.visualize-parameter-sweep.ipynb](hyperlink)|some desc|

**Comment by gwaygenomics** [comment]
> many plots could have font sizes and dot sizes enlarged. they look great on a screen but they don't seem large enough if they were to be a figure on a paper, for instance. but I don't think you should go back to replot everything. It's just something to consider for future plotting.

Definitely agree. I am getting a sense of the landscape now before I dive into a specific story that will require more specific plots.

> plots named in this pattern sample-type_sample-correlation_{TCGA, TARGET, GTEX}_{signal, shuffled}_{spearman, pearson}.png aren't legible. maybe make them bigger...? IMO they are nice quick summary-view of all plots in the subdirectory sample-type. they just aren't very useful in their current forms. maybe make each panel larger? they don't necessarily have to fit the screen.

Yep. They need to expand. Will update in next commit.

> individual panels in sample-type subsubdirectories are good but now the directory tree is becoming really convoluted and deep. this is not essential but you may want to rearrange them by collapsing some of the far-in branches?

Yes, I will consider restructuring.

> also I have noticed that ~30 tissues are represented. you probably explained this somewhere but did the remainder not converge? smaller sample size or any specific reasons?

I am using the column `SMTS` in the GTEx phenotype attributes files. I see that there is also a column (`SMTSD`) that contains more tissue details. I think I will keep the birds eye view of this for now - do think it would be worth getting more detailed now? I plan on it eventually

> I may have said this on Friday. perhaps I am not understanding what you are shuffling or what shuffling is supposed to address. if this 'shuffle' meant to dissociate potential correlations among data to assess null distribution, for instance? If in that line of thoughts, these shuffled training data seem very correlated at z> 10 (e.g. sample-correlation_Adipose Tissue_GTEX_shuffled_pearson.png); was that expected? are these fine-grain latent spaces correlated with anything interpretable? structure among samples perhaps?

I am not surprised by these results. The network can learn _some_ signal, even in largely uncorrelated data. What we want to see are very low correlations in in the testing set. This means that the model is just fitting noise and isn't actually learning anything useful

**Comment by gwaygenomics** [comment]
also @ajlee21 

> Only one question: So it looks like (if I'm understanding this correctly) that you're comparing how well samples correlation across different models (z's) for different datasets. Why are you interested in doing that?

I want to quantify the effect of adding capacity (adding z's) to the compression models. In many sample types, we see that adding a single dimensions increases the ability of the model to explain the sample by a lot. I want to try to explain why (coming later in subsequent PRs!)
